hack/docker-builder/Dockerfile

```diff
@@ -0,0 +1,44 @@
+FROM fedora:28
+
+RUN dnf -y install make git sudo gcc gradle rsync-daemon rsync protobuf-compiler python-pip && \
+    dnf -y clean all
+
+ENV GIMME_GO_VERSION=1.9
+
+RUN mkdir -p /gimme && curl -sL https://raw.githubusercontent.com/travis-ci/gimme/master/gimme | HOME=/gimme bash >> /etc/profile.d/gimme.sh
+
+ENV GOPATH="/go" GOBIN="/usr/bin"
+
+ADD rsyncd.conf /etc/rsyncd.conf
+
+RUN \
+    mkdir -p /go && \
+    source /etc/profile.d/gimme.sh && \
+    go get golang.org/x/tools/cmd/goimports && \
+    git clone https://github.com/mvdan/sh.git $GOPATH/src/mvdan.cc/sh && \
+    cd /go/src/mvdan.cc/sh/cmd/shfmt && \
+    git checkout v2.5.0 && \
+    go get mvdan.cc/sh/cmd/shfmt && \
+    go install && go get -u github.com/onsi/ginkgo/ginkgo && \
+    go get -u -d k8s.io/code-generator/cmd/deepcopy-gen && \
+    go get -u -d k8s.io/code-generator/cmd/defaulter-gen && \
+    go get -u -d k8s.io/code-generator/cmd/openapi-gen && \
+    cd /go/src/k8s.io/code-generator/cmd/deepcopy-gen && \
+    git checkout kubernetes-1.11.2 && \
+    go install && \
+    cd /go/src/k8s.io/code-generator/cmd/defaulter-gen && \
+    git checkout kubernetes-1.11.2 && \
+    go install && \
+    cd /go/src/k8s.io/code-generator/cmd/openapi-gen && \
```

Comment:
Don't hate me Leon, we need to get rid of unneeded dependencies here too. I'm sure it will be easier to go through them now when we have the test working.

Reply:
I could never hate you! I think I've now removed all unused dependencies

Reply:
And it looks so much better now